### PR TITLE
Catch, items and related features

### PR DESCRIPTION
**Pull request by nasserkessas**
See Issue #102.

### CODE REVIEW

**Comment by TurnipGuy30** [comment]
- Item `battle` tags are a brilliant idea. I haven't read over them yet.
- The new-gen trainer types are for use in post-game content, so please don't remove them. You can update the inaccurate ones, though.
- I'll test the new mechanics soon because my laptop just bluescreened... twice.

Nonetheless, thanks for the code!

**Comment by nasserkessas** [comment]
That's ok, no problem.

I'll add the other trainer types back in a sec

**Comment by TurnipGuy30** [comment]
I guess I forgot to mention that as of 09285b0, types automatically colour themselves when using `sp()` or `sg()`.

```python
def reset_sp(speed) -> None:
	global sp, sg
	def sp(text, g=False) -> None:
		for key in colours.keys():
			text = text.replace(f'{key}', f'{colours[key]}{key}{colours["RESET"]}') # this line
		for char in f'{text}\n':
			sleep(speed)
			stdout.write(char)
```

**Comment by nasserkessas** [comment]
No, that's ok, I figured that out by looking at the commit. Did I use the old way by accident or something?

**Comment by TurnipGuy30** [comment]
> No, that's ok, I figured that out by looking at the commit. Did I use the old way by accident or something?

Yeah, in [the response to when a Pokémon is caught](https://github.com/Pokemon-PythonRed/Pokemon-PythonRed/pull/103/files#diff-990f7e4140fab1ad82afc787505090b64d4024e63a891405cd9085e7e6b249ddR392).

> Sorry about the large features, I can imagine they are a pain to review.

It can be, but I do appreciate the help, no need to apologise :D

File: app/main.py [hunk 1]
```diff
@@ -186,6 +186,14 @@ def __init__(self, species, level, ivs, moves=None, chp=None, current_xp=0, fain
 		self.total_xp = xp['total'][self.level_type][str(self.level)] # type: ignore
 		self.current_xp = current_xp
 		self.moves = moves or find_moves(self.species, self.level)
+		self.status = {
+			'paralysis': False,
+			'sleep': False,
+			'freeze': False,
+			'burn': False,
+			'posion': False,
```
Comment (TurnipGuy30) [inline app/main.py:194]: So that's why `self.status['poison']` was throwing a `KeyError`.

**Comment by TurnipGuy30** [comment]
I just realised that "debug" contains "bug"...

image

**Comment by TurnipGuy30** [comment]
When I catch a Pokémon, it seems to be converting my party into `list[dict]`, rather than keeping it as `list[Pokemon]`. This causes commands like `self.check_fainted()` to throw an error.

Would you happen to know why this is the case? I can't find the code that does it. I would recommend using Debug mode.

**Comment by nasserkessas** [comment]
Ok, I'll have a look in a minute

**Comment by nasserkessas** [comment]
@TurnipGuy30, are you sure this occurs when you catch a Pokémon? I see this error when I save the game and start an encounter.

**Comment by TurnipGuy30** [comment]
> ```python
> save_temp = {**save}
> ```

Oh that actually makes sense

**Pull request merged**